pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -529,14 +531,72 @@ async def _completions_create(
                 raise ModelHTTPError(status_code=status_code, model_name=self.model_name, body=e.body) from e
             raise  # pragma: lax no cover
 
+    def _validate_completion(self, response: chat.ChatCompletion) -> chat.ChatCompletion:
+        """Hook that validates chat completions before processing.
+
+        This method may be overridden by subclasses of `OpenAIChatModel` to apply custom completion validations.
+        """
+        return chat.ChatCompletion.model_validate(response.model_dump())
+
+    def _process_reasoning(self, response: chat.ChatCompletion) -> list[ThinkingPart]:
```

Comment:
Let's call this `_process_thinking` for consistent with the `_map_thinking_delta` one below.

Also let's move this below the `_process_response` method